pkg/api/http/directmessaging.go

```diff
@@ -231,14 +231,65 @@ func (a *api) onDirectMessage(w http.ResponseWriter, r *http.Request) {
 
 		w.WriteHeader(int(rResp.Status().GetCode()))
 
-		_, rErr = io.Copy(w, rResp.RawData())
-		if rErr != nil {
-			// Do not return rResp here, we already have a deferred `Close` call on it
-			return nil, backoff.Permanent(rErr)
+		reader := rResp.RawData()
+		flusher, canFlush := w.(http.Flusher)
+
+		accept := r.Header.Get("Accept")
+		isSSE := strings.HasPrefix(accept, "text/event-stream")
```

Comment:
Does it have to be a prefix?
```suggestion
		isSSE := strings.Contains(accept, "text/event-stream")
```